core/consensus/grandpa/impl/launcher_impl.cpp

```diff
@@ -177,20 +180,50 @@ namespace kagome::consensus::grandpa {
   void LauncherImpl::onVoteMessage(const VoteMessage &msg) {
     auto current_round = current_round_;
     auto current_round_number = current_round->roundNumber();
-    // TODO(xDimon): Implement mechanism to check if signer is known
-    if (msg.round_number == current_round_number) {
-      visit_in_place(
-          msg.vote.message,
-          [&current_round, &msg](const PrimaryPropose &primary_propose) {
-            current_round->onPrimaryPropose(msg.vote);
-          },
-          [&current_round, &msg](const Prevote &prevote) {
-            current_round->onPrevote(msg.vote);
-          },
-          [&current_round, &msg](const Precommit &precommit) {
-            current_round->onPrecommit(msg.vote);
-          });
+
+    // ensure we are in current round
+    if (msg.round_number != current_round_number) {
```

Comment:
shouldn't we log it?

Reply:
No, it's regular case. Some votes can late during target round.